content/2017.04.getting-started-with-elm.rst

```diff
@@ -0,0 +1,487 @@
+Getting started with Elm
+########################
+
+:lang: en
+:date: 2017-04-13
+:url: getting-started-with-elm
+:summary: What is Elm? how does it work? How can I try it really fast?
+
+
+Introduction
+============
+
+The past two weeks we have been traveling and meeting a bunch of
+developers.
+
+It was a chance for us to tell them about Elm through talks at
+conferences and to present a short one-hour workshop.
+
+You can find the slides of the two talks here:
+
+- `a short introduction to Elm by Niko <http://slides.com/n1k0/elm>`_
+- `Talking JSON with Elm by Natim <http://natim.github.io/django-slides/elm-json/>`_
+
+However, this article is a summary of the take aways of this workshop
+for those who would like to know a bit more about Elm.
+
+
+What is Elm and why should I be interested?
+===========================================
+
+Elm is a toolkit that is focused on making sure your app state and the
+HTML that reflects it are always in sync.
+
+If you have been doing any frontend development during the last couple
+of years, you might have seen that the JavaScript ecosystem is a bit
+wild. Keeping up-to-date with the growing number of competing
+frameworks can take a lot of energy, not to mention trying your app on
+all the available browser versions to make sure it works well
+everywhere. Training a team to build an app with React.js,
+Angular2.js, Vue.js or Ember.js as well as looking for the right
+boilerplate to get started with your project are good examples of how
+hight the frustration that the constant reinvention of the ecosystem
+causes can be.
+
+On the other hand, writing tests for the front-end can be tedious and
+supporting Flow or TypeScript can prove to be yet another layer to
+handle in your stack.
+
+If you are already up-to-speed with all those stacks, then you should
+probably keep using them. However if you don't know which one to
+choose and you are looking for a painless way to build frontend
+applications, read on.
+
+
+How is Elm different?
+=====================
+
+First of all, using Elm, the two lines it takes to load your app is
+all the JavaScript you will write. Your application will be written in
+Elm and compiled into JavaScript.
+
+All the mistakes you will do will be caught at compilation, which
+means that you will not have errors when running your app in the
+browser.
+
+Because you will have a compilation step, Elm will force you to handle
+every possible case of your model state. It will make sure you covered
+them on the rendering side.
```

Comment:
I'd probably put an example here, which may be enlightening to some:

```js
function foo(bar) {
  return bar.baz;
}
foo(null); // explodes
```

vs

```elm
type alias Bar = {
  baz : String
}

foo : Maybe Bar -> String
foo bar = 
  case bar of
    Nothing -> "" -- we're forced to deal with a possibly null value
    Just bar -> bar.baz
```

(or smthg like that)